nemo/lightning/resume.py

```diff
@@ -1,183 +1,209 @@
-    """Class that handles the logic for setting checkpoint paths and restoring from
-    checkpoints in NeMo.
-    """
-
-    def __init__(
-        self,
-        path: Optional[str] = None,  ## old resume_from_checkpoint
-        dirpath: Optional[str] = None,  ## optional path to checkpoint directory
-        import_path: Optional[str] = None,  ## for importing from hf or other checkpoint formats
-        adapter_path: Optional[str] = None,
-        resume_if_exists: bool = False,
-        resume_past_end: bool = False,
-        resume_ignore_no_checkpoint: bool = False,
-    ):
-        """
-        Args:
-            path (str): Can be used to specify a path to a specific checkpoint file to load from.
-                This will override any checkpoint found when resume_if_exists is True.
-                Defaults to None
-            dirpath (str): Path to the checkpointing directory to restore from. Defaults to <log_dir>/checkpoints
-            import_path (str): Path to specify if importing a checkpoint from HF or
-                another non-NeMo checkpoint format. If import_path is provided, other arguments
-                are unused.
-            adapter_path (str): Path to any adapter checkpoints.
-            resume_if_exists (bool): Whether this experiment is resuming from a previous run. If
-                True, it sets trainer._checkpoint_connector._ckpt_path so that the trainer should
-                auto-resume. exp_manager will move files under log_dir to log_dir/run_{int}.
-                Defaults to False.
-            resume_past_end (bool): By default, AutoResume throws an error if resume_if_exists is
-                True and a checkpoint matching ``*end.ckpt`` indicating a previous training run
-                fully completed. Setting resume_past_end=True disables this behavior and loads the
-                last checkpoint.
-            resume_ignore_no_checkpoint (bool): AutoResume throws an error if resume_if_exists is
-                True and no checkpoint could be found. Setting resume_ignore_no_checkpoint=True
-                disables this behavior, in which case exp_manager will print a message and
-                continue without restoring.
-        """
-        if path and import_path:
-            raise ValueError("Only one of path or import_path can be set")
-
-        self.path = path
-        self.dirpath = dirpath
-        self.import_path = import_path
-        self.adapter_path = adapter_path
-        self.resume_if_exists = resume_if_exists
-        self.resume_past_end = resume_past_end
-        self.resume_ignore_no_checkpoint = resume_ignore_no_checkpoint
-
-    def nemo_path(self, model=None) -> Optional[Path]:
+        trainer_ckpt_path = self.get_trainer_ckpt_path(model)
+        if trainer_ckpt_path:
+            trainer.ckpt_path = trainer_ckpt_path
+            trainer.checkpoint_callback.last_model_path = trainer_ckpt_path
+        elif self.selective_restore_config:
```

Comment:
I think there's a bug here, my suggestion is to change `elif` to `if`.

Reply:
This is intended as you cannot have both together. Basically, you want to skip selective restore if checkpoint path is found.

Reply:
Raising an error would not support the following usecase:
1. Start finetuning from SelectiveRestoreConfig
2. Resume finetuning from the last saved checkpoint by 1.